File: gym/elems/00_00_sample_parity/cpp/parity_test.cc
#include "parity.h"
#include <cassert>
#include <iostream>

int main() {
    assert(parity::ComputeParity(0b1011) == 1);
    assert(parity::ComputeParity(0b10001) == 0);
    assert(parity::ComputeParity(0) == 0);
    std::cout << "All C++ parity tests passed!" << std::endl;
    return 0;
}

File: gym/elems/00_00_sample_parity/go/parity_test.go
package parity

import "testing"

func TestComputeParity(t *testing.T) {
	tests := []struct {
		input uint64
		want  int16
	}{
		{0b1011, 1},
		{0b10001, 0},
		{0, 0},
	}

	for _, tt := range tests {
		got := ComputeParity(tt.input)
		if got != tt.want {
			t.Errorf("ComputeParity(%b) = %d; want %d", tt.input, got, tt.want)
		}
	}
}
